code/modules/admin/admin.dm

```diff
@@ -661,23 +639,45 @@
 
 /datum/admins/proc/delay()
 	set category = "Server"
-	set desc="Delay the game start"
-	set name="Delay pre-game"
-
-	var/newtime = input("Set a new time in seconds. Set -1 for indefinite delay.","Set Delay",round(SSticker.GetTimeLeft()/10)) as num|null
-	if(SSticker.current_state > GAME_STATE_PREGAME)
-		return alert("Too late... The game has already started!")
-	if(newtime)
-		newtime = newtime*10
-		SSticker.SetTimeLeft(newtime)
-		if(newtime < 0)
-			to_chat(world, "<b>The game start has been delayed.</b>")
-			log_admin("[key_name(usr)] delayed the round start.")
+	set desc="Delay the game"
+	set name="Delay Start/End"
+
+	if(!check_rights(R_ADMIN)) //YOGS - R_SERVER -> R_ADMIN
+		return
+
+	if(SSticker.current_state < GAME_STATE_PLAYING)
+		var/newtime = input("Set a new time in seconds. Set -1 for indefinite delay.","Set Delay",round(SSticker.GetTimeLeft()/10)) as num|null
+		if(newtime)
+			newtime = newtime*10
+			SSticker.SetTimeLeft(newtime)
+			if(newtime < 0)
+				to_chat(world, "<b>The game start has been delayed.</b>")
+				log_admin("[key_name(usr)] delayed the round start.")
+				message_admins("[key_name(usr)] delayed the round start.")
+			else
+				to_chat(world, "<b>The game will start in [DisplayTimeText(newtime)].</b>")
+				SEND_SOUND(world, sound('sound/ai/attention.ogg'))
+				message_admins("[key_name(usr)] set the pre-game delay to [DisplayTimeText(newtime)].")
+				log_admin("[key_name(usr)] set the pre-game delay to [DisplayTimeText(newtime)].")
+			SSblackbox.record_feedback("tally", "admin_verb", 1, "Delay Game Start") //If you are copy-pasting this, ensure the 2nd parameter is unique to the new proc!
+
+	else if(SSticker.current_state >= GAME_STATE_PLAYING)
+		if(!SSticker.delay_end)
+			SSticker.admin_delay_notice = input(usr, "Enter a reason for delaying the round end", "Round Delay Reason") as null|text
+			if(isnull(SSticker.admin_delay_notice))
+				return
 		else
-			to_chat(world, "<b>The game will start in [DisplayTimeText(newtime)].</b>")
-			SEND_SOUND(world, sound('sound/ai/attention.ogg'))
-			log_admin("[key_name(usr)] set the pre-game delay to [DisplayTimeText(newtime)].")
-		SSblackbox.record_feedback("tally", "admin_verb", 1, "Delay Game Start") //If you are copy-pasting this, ensure the 2nd parameter is unique to the new proc!
+			SSticker.admin_delay_notice = null
+		SSticker.delay_end = !SSticker.delay_end
+		var/reason = SSticker.delay_end ? "for reason: [SSticker.admin_delay_notice]" : "."//laziness
```

Comment:
``Having it insert a dot would display X has delayed the round end .`` with that space at the end
```suggestion
		var/reason = SSticker.delay_end ? "for reason: [SSticker.admin_delay_notice]" : ""
```